### ISSUE TITLE
Need pass the plain text value

### ISSUE THREAD

**Issue by hyrious** [markedjs/marked-gfm-heading-id#225]
As described in the [github-slugger](https://github.com/Flet/github-slugger#readme) readme, you should pass in plain text to this library, these examples will lead to differences to GitHub:

```markdown
# Hello **world!**

# <samp>Hello <ins>world!</ins></samp>
```

**Comment by hyrious** [comment]
Wait, maybe I was wrong..

**Comment by hyrious** [comment]
Okay I think I know what's wrong here, the `raw` param in `heading(text, level, raw)` is correct but it does not handle user input html literals:

```markdown
# <samp>Hello <ins>world!</ins></samp>
```

Which means this example will not work. A simple solution would be using a simple regex to strip these html tags:

```js
slugger.slug(raw.replace(/<[^>]+>/g, ''))
```

**Comment by UziTech** [comment]
This is probably something that should be fixed in marked instead of in this package. `raw` should be the text that is used to determine the slug.

**Comment by UziTech** [comment]
it actually looks like marked does this in Slugger with the line `.replace(/<[!\/a-z].*?>/ig, '')`

**Comment by hyrious** [comment]
You're right.

**Comment by UziTech** [comment]
I added a fix and a bunch of tests in v2.0.2
